app/components/FileCard.tsx

```diff
@@ -54,23 +51,38 @@ const FileCard = (props: FileCardProps) => {
         }
     }, [props.fileType, props.fileURL]);
 
-    const fetchImage = () => {
-        fetchFile(props.fileURL)
-            .then(base64Image => {
-                if (base64Image) {
-                    setPreviewImage({ uri: base64Image });
-                } else {
-                    console.error('Failed to fetch image: No data received.');
-                }
-            })
-            .catch(error => {
-                console.error(error);
-            });
+    const fetchImage = async () => {
+        const cacheDirectory = FileSystem.cacheDirectory + "images/";
+        const fileName = props.fileURL.split("/").pop();
+        const fileUri = cacheDirectory + fileName;
```

Comment:
So I got the following error:
```
[Error: Call to function 'ExponentFileSystem.getInfoAsync' has been rejected.
→ Caused by: java.io.IOException: Location '/data/user/0/host.exp.exponent/cache/images/15:30T020724.jpeg' isn't readable.]
```
And I noticed that it is because the fileName of images can contain ":", eg. in "19:045T02-07-24.jpeg", which seems to be an illegal character to use when creating a file/folder. Did it work for you anyways?

But the solution seems simple: ``fileName?.replaceAll(":", "")``
So removing the colon.
But of course there could be other forbidden characters, therefore we should hash the entire string. I found this function on [StackOverflow](https://stackoverflow.com/a/40958826/8909353):
```js
function hashString(input: string) {
        var hash = 0, len = input.length;
        for (var i = 0; i < len; i++) {
            hash  = ((hash << 5) - hash) + input.charCodeAt(i);
            hash |= 0;
        }
        return hash;
    }
```

Maybe put it in utils/utils.js and use it to create a hash from the fileName, then use it here like this for example:
```js
const fileUri = cacheDirectory + fileName;
```